genai/tools/tools_code_exec_with_txt_local_img.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package tools shows examples of various tools that Gemini model can use to generate text.
+package tools
+
+// [START googlegenaisdk_tools_code_exec_with_txt_local_img]
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	genai "google.golang.org/genai"
+)
+
+// generateWithLocalImgAndCodeExec shows how to combine a local image, a text prompt,
+// and enable the code execution tool in a request.
+func generateWithLocalImgAndCodeExec(w io.Writer) error {
+	ctx := context.Background()
+
+	client, err := genai.NewClient(ctx, &genai.ClientConfig{
+		HTTPOptions: genai.HTTPOptions{APIVersion: "v1"},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create genai client: %w", err)
+	}
+
+	// Read local image
+	imgBytes, err := os.ReadFile("640px-Monty_open_door.svg.png")
+	if err != nil {
+		return fmt.Errorf("failed to read image: %w", err)
+	}
+
+	// Define the prompt
+	prompt := "Run a simulation of the Monty Hall Problem with 1,000 trials.\n" +
+		"Here's how this works as a reminder. In the Monty Hall Problem, you're on a game\n" +
+		"show with three doors. Behind one is a car, and behind the others are goats. You\n" +
+		"pick a door. The host, who knows what's behind the doors, opens a different door\n" +
+		"to reveal a goat. Should you switch to the remaining unopened door?\n" +
+		"The answer has always been a little difficult for me to understand when people\n" +
+		"solve it with math - so please run a simulation with Python to show me what the\n" +
+		"best strategy is.\n" +
+		"Thank you!"
```

Comment:
issue: Choppy multi-line string reading experience
suggestion: Could we use a multi-line string to streamline reading the prompt? Not sure if there is an explicit pattern in place for genai samples.